src/algebra/lattice_ordered_group.lean

```diff
@@ -0,0 +1,607 @@
+(mul_le_mul_left : ∀ a b : α, a ≤ b → ∀ c : α, c * a ≤ c * b)
+
+/--
+A group `(α,+)`, equipped with a partial order `≤`, making `α` into a lattice, such that, given
+elements `a` and `b` of type `α` satisfying `a ≤ b`, for all elements `c` and `d` of type `α`,
+$$c + a + d ≤ c + b + d,$$
+is said to be a *lattice ordered (additive) group*.
+-/
+class lattice_ordered_add_group (α : Type u)
+  extends add_group α, lattice α :=
+(add_le_add : ∀ a b : α, a ≤ b → ∀ c d : α, c + a + d ≤ c + b + d)
+
+/--
+A lattice ordered additive group with Abelian group addition is said to be a *lattice ordered
+(additative) commutative group*.
+-/
+class lattice_ordered_add_comm_group (α : Type u)
+  extends add_comm_group α, lattice α :=
+(add_le_add_left : ∀ a b : α, a ≤ b → ∀ c : α, c + a ≤ c + b)
+
+-- Tells to_additive that lattice_ordered_add_group is the additive form of lattice_ordered_group
+-- and lattice_ordered_add_comm_group is the additive form of lattice_ordered_comm_group
+attribute [to_additive] lattice_ordered_group
+attribute [to_additive] lattice_ordered_comm_group
+
+-- Every lattice ordered commutative group is also an ordered additive commutative group
+@[priority 100, to_additive] -- see Note [lower instance priority]
+instance lattice_ordered_comm_group.to_ordered_comm_group (α : Type u)
+  [s : lattice_ordered_comm_group α] : ordered_comm_group α :=
+{ mul := s.mul, ..s }
+
+-- A lattice ordered commutative group is a lattice ordered group
+@[priority 100, to_additive] -- see Note [lower instance priority]
+instance lattice_ordered_comm_group.to_lattice_ordered_group (α : Type u)
+  [s : lattice_ordered_comm_group α] : lattice_ordered_group α :=
+{ mul_le_mul :=
+  begin
+    rintros a b h c d,
+    rw [mul_assoc, mul_assoc],
+    apply s.mul_le_mul_left,
+    rw mul_comm,
+    nth_rewrite 1 mul_comm,
+    apply s.mul_le_mul_left,
+    exact h,
+  end }
+
+
+-- A linearly ordered additive commutative group is a lattice ordered commutative group
+@[priority 100, to_additive] -- see Note [lower instance priority]
+instance linear_ordered_comm_group.to_lattice_ordered_comm_group (α : Type u)
+ [s : linear_ordered_comm_group α] : lattice_ordered_comm_group α :=
+{ mul_le_mul_left := s.mul_le_mul_left, }
+
+variables {α : Type u} [lattice_ordered_add_comm_group α]
+
+-- Special case of Bourbaki A.VI.9 (1)
+/--
+Let `α` be a lattice ordered commutative group. For all elements `a`, `b` and `c` in `α`,
+$$c + (a⊔b) = (c+a)⊔(c+b).$$
+-/
+lemma add_join_eq_add_join_add (a b c : α) : c + (a⊔b) = (c+a)⊔(c+b) :=
+begin
+  rw le_antisymm_iff,
+  split,
+  { rw [← add_le_add_iff_left (-c), ← add_assoc, neg_add_self, zero_add, sup_le_iff],
+    split,
+    { simp, },
+    { simp, } },
+  { rw ← add_le_add_iff_left (-c), simp,  }
+end
+
+-- Special case of Bourbaki A.VI.9 (2)
+/--
+Let `α` be a lattice ordered commutative group. For all elements `a` and `b` in `α`,
+$$-(a⊔b)=(-a)⊓(-b).$$
+-/
+lemma neg_join_eq_neg_meet_neg (a b : α) : -(a⊔b)=(-a)⊓(-b) :=
```

Comment:
Lemma names should use `sup`  instead of `join` and `inf` instead of `meet`.

Reply:
Changed, thanks.